### PR TITLE
enhance button with icon

### LINKED ISSUE #28: Enhance `Button with Icon` component
**Goal**
We wanted to add some small changes to the `button with icon` component.

**Tasks**

- [x] The Icon Size should always match the text size.
- [x] The Icon color should be a lighter (or darker, whatever works better) of the button theme.

### PR DESCRIPTION
**Pull request by DaanSchar**
closes #28

### CODE REVIEW

**Comment by DaanSchar** [comment]
i took a different approach here, since i didnt get it to work with svg in the style

File: src/components/common/ButtonWithIcon.vue [hunk 1]
```diff
@@ -5,7 +5,12 @@
     :size="size"
   >
     <div class="flex items-center">
-      <slot name="icon" />
+      <div
+        :class="classes"
+        class="mr-1"
```
Comment (tsuukii) [inline src/components/common/ButtonWithIcon.vue:10]: I would prefer a bit of a larger margin. What do you think of `mr-2`?

File: src/components/common/ButtonWithIcon.vue [hunk 2]
```diff
@@ -33,6 +38,29 @@ export default {
       type: String,
       default: '',
     },
+  },
+  computed: {
+    classes() {
+      return `${this.disabled ? 'disabledIcon' : this.type + 'Icon'} ${this.size}}`
```
Comment (tsuukii) [inline src/components/common/ButtonWithIcon.vue:44]: I actully had a error in my computed property for the base button. It has been fixed in the main branch but you still have the error (one `}` too much at the end of the line).

```suggestion
      return `${this.disabled ? 'disabledIcon' : this.type + 'Icon'} ${this.size} `
```

File: src/components/common/ButtonWithIcon.vue [hunk 3]
```diff
@@ -33,6 +38,29 @@ export default {
...
+    },
   }
 }
 </script>
+
+<style scoped>
+.primaryIcon{
```
Comment (tsuukii) [inline src/components/common/ButtonWithIcon.vue:51]: I wouldn't create new class names like this. You can just use `.primary svg` as we are already assigning it in this way. See more here: https://github.com/Galleme/frontend/issues/28#issuecomment-917533657

File: src/stories/common/ButtonWithIcon.stories.js [hunk 4]
```diff
@@ -26,13 +26,37 @@ const Template = (args) => ({
...
+                <fa :icon="faFlag"/>
             </template>
         </button-with-icon>`,
 });
 
 export const Primary = Template.bind({});
 Primary.args = {
     type: 'primary',
-    size: 'text-md',
+    size: 'text-sm ',
```
Comment (tsuukii) [inline src/stories/common/ButtonWithIcon.stories.js:37]: With the fix I mentioned above. Please remove the space after `text-sm`. Do the same for all the following storybook examples are well please.

**Review by zoey-kaiser — CHANGES_REQUESTED** [review]

**Comment by tsuukii** [comment]
Your way of doing it works too. I had tested it a bit yesterday and I have gotten the solution with `.primary svg` to work, however I think creating new classes would work too, so its fine :)

**Review by zoey-kaiser — APPROVED** [review]

**Pull request merged**